huaweicloud/services/cbr/data_source_huaweicloud_cbr_external_vaults.go

```diff
@@ -0,0 +1,522 @@
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"value": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataExternalVaultResourcesSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protect_status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"backup_size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"backup_count": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"extra_info": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     dataExternalVaultResourcesExtraInfoSchema(),
+			},
+		},
+	}
+}
+
+func dataExternalVaultResourcesExtraInfoSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"exclude_volumes": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func dataExternalVaultBillingSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"allocated": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"used": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"charging_mode": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"cloud_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"consistent_level": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protect_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"object_type": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"spec_code": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"order_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"product_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"storage_unit": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"frozen_scene": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"is_multi_az": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func buildQueryParams(d *schema.ResourceData) string {
+	queryParams := ""
+	if v, ok := d.GetOk("external_project_id"); ok {
+		queryParams = fmt.Sprintf("%s&external_project_id=%v", queryParams, v)
+	}
+	if v, ok := d.GetOk("region_id"); ok {
+		queryParams = fmt.Sprintf("%s&region_id=%v", queryParams, v)
+	}
```

Comment:
Fields `external_project_id` and `region_id` are both required, they always exists. It is not necessary to judge whether they are exists.